drivers/display/ls0xx.c

```diff
@@ -0,0 +1,501 @@
+ * -> If devicetree contains DISP pin then it will be set to high
+ *      during initialization.
+ * -> Sharp memory displays require toggling the VCOM signal periodically
+ *      to prevent a DC bias ocurring in the panel. The DC bias can damage
+ *      the LCD and reduce the life. This signal must be supplied
+ *      from either serial input (sw) or an external signal on the
+ *      EXTCOMIN pin. The driver can do this internally
+ *      (CONFIG_LS0XX_VCOM_DRIVER=y) else user can handle it in
+ *      application code (CONFIG_LS0XX_VCOM_DRIVER=n).
+ *      Source:
+ * https://www.sharpsde.com/fileadmin/products/Displays/2016_SDE_App_Note_for_Memory_LCD_programming_V1.3.pdf
+ * -> If CONFIG_LS0XX_VCOM_DRIVER=y then driver can partially or completely
+ *      handle VCOM toggling. User can control method of toggling
+ *      using CONFIG_LS0XX_VCOM_EXTERNAL.
+ * -> VCOM is toggled through serial input by software by default
+ *      CONFIG_LS0XX_VCOM_EXTERNAL=n
+ *      EXTMODE pin is connected to VSS
+ *      Important: User has to make sure .write is called periodically
+ *      for toggling VCOM. If there is no data to update then buf can
+ *      be set to NULL then only VCOM will be toggled.
+ * -> To control VCOM using external signal EXTCOMIN, set
+ *      CONFIG_LS0XX_VCOM_EXTERNAL=y
+ *      EXTMODE pin is connected to VDD
+ *      Important: Driver will start a thread which will
+ *      toggle the EXTCOMIN pin every 500ms. There is no
+ *      dependency on user.
+ * -> line_buf format for each row.
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+
+#define LS0XX_SCS_PIN             DT_INST_GPIO_PIN(0, scs_gpios)
+#define LS0XX_SCS_CNTRL           DT_INST_GPIO_LABEL(0, scs_gpios)
+
+#if DT_INST_NODE_HAS_PROP(0, disp_gpios)
+	#define LS0XX_DISP_PIN        DT_INST_GPIO_PIN(0, disp_gpios)
+	#define LS0XX_DISP_CNTRL      DT_INST_GPIO_LABEL(0, disp_gpios)
+#endif
+
+#if defined(CONFIG_LS0XX_VCOM_DRIVER)
+	#if defined(CONFIG_LS0XX_VCOM_EXTERNAL) && DT_INST_NODE_HAS_PROP(0, extcomin_gpios)
+		#define LS0XX_EXTCOMIN_PIN    DT_INST_GPIO_PIN(0, extcomin_gpios)
+		#define LS0XX_EXTCOMIN_CNTRL  DT_INST_GPIO_LABEL(0, extcomin_gpios)
+	#endif
+#endif
+
+#define LS0XX_PANEL_WIDTH         DT_INST_PROP(0, width)
+#define LS0XX_PANEL_HEIGHT        DT_INST_PROP(0, height)
+
+#define LS0XX_PIXELS_PER_BYTE     8U
+/* Adding 2 for the line number and dummy byte */
+#define LS0XX_BYTES_PER_LINE      ((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE)+2)
+
+#define LS0XX_BIT_WRITECMD        0x01
+#define LS0XX_BIT_VCOM            0x02
+#define LS0XX_BIT_CLEAR           0x04
+
+static uint8_t line_buf[LS0XX_BYTES_PER_LINE];
+
+static struct spi_buf tx_buf;
+static struct spi_buf_set buf_set = {.buffers = &tx_buf, .count = 1};
+
+struct ls0xx_data {
+	struct device *scs;
+#if defined(LS0XX_DISP_CNTRL)
```

Comment:
Do not obfuscate where it comes from, change to #if DT_INST_NODE_HAS_PROP(0, disp_gpios), the same below.

I also thing we should reconsider this practice, sacrifice these few bytes and remove all _#if_ and use` IS_ENABLED`.